packages/cbioportal-utils/src/mutation/MutationAnnotator.ts

```diff
@@ -324,12 +322,23 @@ export function findCanonicalTranscript(
 export function indexAnnotationsByGenomicLocation(
     variantAnnotations: VariantAnnotation[]
 ): { [genomicLocation: string]: VariantAnnotation } {
-    return _.keyBy(variantAnnotations, annotation =>
-        annotation.annotation_summary
-            ? genomicLocationString(
-                  annotation.annotation_summary.genomicLocation
-              )
-            : genomicLocationStringFromVariantAnnotation(annotation)
+    return _.reduce(
+        variantAnnotations,
+        (indexAnnotationsByGenomicLocation, annotation) => {
+            let key = annotation.annotation_summary
+                ? genomicLocationString(
+                      annotation.annotation_summary.genomicLocation
+                  )
+                : genomicLocationStringFromVariantAnnotation(annotation);
+            indexAnnotationsByGenomicLocation[key] = annotation;
+            if (key !== annotation.originalVariantQuery) {
+                indexAnnotationsByGenomicLocation[
+                    annotation.originalVariantQuery
+                ] = annotation;
+            }
+            return indexAnnotationsByGenomicLocation;
```

Comment:
@leexgh could you add some comments in the code that describe what you mentioned in the PR? From what I understand this is mostly to avoid changing the logic in other places. Using genomicLocation to map back won't always be correct, so don't you think it's best to use `originalVariantQuery` everywhere? E.g. we could use the annotation endpoint instead to get the hotspots. If it's too much work to change everywhere, one option is to add a `// TODO:` explaining what needs to be done before we can use `originalVariantQuery` everywhere

Reply:
@inodb I see, I found we also use genomic location to query myVariantInfo so I did this temp fix to avoid too much code changes. But yes using `originalVariantQuery` is more clear. I'm not exactly sure how many places we need to change, shouldn't be too much I guess. I'll try to change them all using annotation endpoint and mapping with `originalVariantQuery`